src/main/java/models/ShoppingCart.java

```diff
@@ -34,8 +36,7 @@ public ShoppingCart(){ }
 
     public Sale checkout() {
         Sale sale = new Sale(this.books, this.customer);
-        Bookstore currBookstore;
-        for (Book book : books){
+        for (Book book : this.books){
```

Comment:
was currBookstore not used?

Reply:
Nope. Must've been left over from a different for loop or something.